pkgs/desktops/gnome-3/extensions/default.nix

```diff
@@ -0,0 +1,25 @@
+{pkgs, ...}: let
+  # Index of all scraped extensions (with supported versions)
+  extensionsIndex = builtins.fromJSON (builtins.readFile ./extensions.json);
+  # Filter the extensions index to the matching Gnome shell version.
+  filterShellVersion = shell-version: extensions-index:
+    (builtins.map (e: {
```

Comment:
`map` is in the default prelude no need to use `builtins`.

What is `e`? Can we give it a name?

Maybe this entire function that is passed to `map` should have a name & a docstring.